### ISSUE TITLE
Improve the robustness of the node cordoner

### ISSUE THREAD

**Issue by choldgraf** [jupyterhub/binderhub#566]
[gitter chat link](https://gitter.im/jupyterhub/binder?at=5afc705041f54f22e234c279)

We recently deployed a cordoning step to the image cleaner, so that launches don't error on the node while inodes are being cleaned out.

However, I noticed that we're getting spotty behavior on how the cordoning works. For example, see these three images:

node cordoning data shows three cordoning events, but two that were uncordoned immediately (even though inode cleanup still happened)
![image](https://user-images.githubusercontent.com/1839645/40185686-09d56b38-59a8-11e8-92cc-43209aa982d7.png)

inodes plot shows that we have 3 inodes cleanup sessions
![](https://files.gitter.im/jupyterhub/binder/xP8U/thumb/image.png)

Launch success % shows that launches fail only when the cordoning failed to occur
![](https://files.gitter.im/jupyterhub/binder/s6mE/thumb/image.png)

Looking into the logs, there were also times that the image cleaner _said_ it was cordoning the node, but this never seemed to appear in the plots. E.g., we got logs like this:
```
79.6% inodes used
79.9% inodes used
80.1% inodes used
408 images available to prune
Cordoning node gke-prod-a-ssd-pool-32-134a959a-cvh2
Removing gcr.io/binder-prod/r2d-05168b0-gibiansky-2dihaskell-790c32:3cd40d186c800b3360761e312d8e9ddacc83c994 (size=10.94GB)
Timeout removing gcr.io/binder-prod/r2d-05168b0-gibiansky-2dihaskell-790c32:3cd40d186c800b3360761e312d8e9ddacc83c994
Removing gcr.io/binder-prod/r2d-05168b0-neuropoly-2dqmrlab-043064:ca5a1465de8b7a1ee5cfd939fb779e34affc87e6 (size=8.25GB)
Failed to remove gcr.io/binder-prod/r2d-05168b0-neuropoly-2dqmrlab-043064:ca5a1465de8b7a1ee5cfd939fb779e34affc87e6, skipping this image
409 Client Error: Conflict ("conflict: unable to delete 80ffe9dc4fda (cannot be forced) - image is being used by running container 9173853c81fd")
Removing gcr.io/binder-prod/r2d-05168b0-annefou-2dmetos-5fpython-77861a:267a7563f3d427faa53fad54f8a3d7bf3887fa32 (size=7.98GB)
...
...etc
...
Removing gcr.io/binder-prod/r2d-05168b0-rlabbe-2dkalman-2dand-2dbayesian-2dfilters-2din-2dpython-0be055:b00adb2786a175e8aa67c2411612f8ee6613c3d9 (size=3.12GB)
Removed gcr.io/binder-prod/r2d-05168b0-rlabbe-2dkalman-2dand-2dbayesian-2dfilters-2din-2dpython-0be055:b00adb2786a175e8aa67c2411612f8ee6613c3d9
Removing gcr.io/binder-prod/r2d-05168b0-sourceryinstitute-2djupyter-2dcaf-2dkernel-5cc4b3:691f464b5a4bc4a37d1e187b5ab37e08f7b323bb (size=3.11GB)
Removed gcr.io/binder-prod/r2d-05168b0-sourceryinstitute-2djupyter-2dcaf-2dkernel-5cc4b3:691f464b5a4bc4a37d1e187b5ab37e08f7b323bb
Uncordoning node gke-prod-a-ssd-pool-32-134a959a-cvh2
59.7% inodes used
59.7% inodes used
59.7% inodes used
```
but history shows no cordoning/uncordoning of the node.

I'm pasting this info here so we can keep track of this behavior and try to smooth it out. When the cordoning happens properly, we get through inodes cleanup with almost no down time!

cc @minrk @betatim as they've been involved heavily with this so far!

**Comment by betatim** [comment]
If I cordon a node by hand with `kubectl cordon <node>` it shows up on the grafana plot. This makes me think that the plotting works.

Another observation: inode cleaning doesn't always lead to a dip in launch success. But when we clean inodes on the node that hosts the hub pod it does seem to lead to a dip.

**Comment by choldgraf** [comment]
this one is closed now by #568 !
